SoftwareGuide/Latex/Appendices/CodingStyleGuide.tex

```diff
@@ -1428,6 +1430,27 @@ \section{Pipelines}
 
 \end{itemize}
 
+\section{The auto Keyword}
+\label{sec:TheAutoKeyword}
+
+Available since C++11, the \code{auto} keyword specifies that a variables type
+is automatically deduced from its initializer.
+
+The \code{auto} keyword should be used to specify a type in the following cases:
+
+\begin{itemize}
+\item The type is duplicated on the left side of an assigment when it is mandated on the
+      right side, i.e. when there is an explicit cast.
+\item When obtaining container elements, when the element type is obvious from the type
+      of the container.
+\item When declaring iterators in a \code{for} loop.
+\item When initializing with new, but not ITK's \code{::New()}.
+\end{itemize}
+
+All other cases \textbf{should not} use \code{auto}, but a semantic type name
+should be used that convey's meaning, as described in Section~\ref{sec:NamingConventions}
+and Section~\ref{subsec:NamingTypedefs}.
+
```

Comment:
should not is quite strong @thewtex! What do you think of:

When the type isn't clear from the context \code{auto localValue = itkFilter->GetOutput()}, prefer to write the descriptive type \code{ScalatMetricType localValue = itkFilter->GetOutput()}, or provide a descriptive variable name if using auto \code{auto scalarResult =itkFilter->GetOutput()}

Reply:
"The type isn't clear from the context" is ambiguous -- it could always be argued that it is already clear from the context. The point of the style guide and a consistent adherence is to have a consistent style throughout the toolkit and to avoid spending time debating over style choices in every patch.

Reply:
Removing the unnecessary bold font emphasis on *should not*.